epochStart/metachain/economics_test.go

```diff
@@ -442,6 +443,161 @@ func TestEconomics_VerifyRewardsPerBlock_DifferentHitRates(t *testing.T) {
 	}
 }
 
+func TestComputeEndOfEpochEconomics(t *testing.T) {
+	t.Parallel()
+
+	commAddress := "communityAddress"
+	totalSupply, _ := big.NewInt(0).SetString("20000000000000000000000000000", 10) // 20 Billions ERD
+	nodePrice, _ := big.NewInt(0).SetString("1000000000000000000000", 10)          // 1000 ERD
+
+	roundDur := 4
+	args := getArguments()
+	args.RewardsHandler = &mock.RewardsHandlerStub{
+		MaxInflationRateCalled: func() float64 {
+			return 0.1
+		},
+		CommunityAddressCalled: func() string {
+			return commAddress
+		},
+		CommunityPercentageCalled: func() float64 {
+			return 0.1
+		},
+		LeaderPercentageCalled: func() float64 {
+			return 0.1
+		},
+	}
+	args.RoundTime = &mock.RoundTimeDurationHandler{
+		TimeDurationCalled: func() time.Duration {
+			return time.Duration(roundDur) * time.Second
+		},
+	}
+	hdrPrevEpochStart := block.MetaBlock{
+		Round: 0,
+		Nonce: 0,
+		Epoch: 0,
+		EpochStart: block.EpochStart{
+			Economics: block.Economics{
+				TotalSupply:            totalSupply,
+				TotalToDistribute:      big.NewInt(10),
+				TotalNewlyMinted:       big.NewInt(10),
+				RewardsPerBlockPerNode: big.NewInt(10),
+				NodePrice:              nodePrice,
+				RewardsForCommunity:    big.NewInt(10),
+			},
+			LastFinalizedHeaders: []block.EpochStartShardData{
+				{ShardID: 0, Nonce: 0},
+				{ShardID: 1, Nonce: 0},
+			},
+		},
+	}
+	args.Store = &mock.ChainStorerStub{
+		GetStorerCalled: func(unitType dataRetriever.UnitType) storage.Storer {
+			// this will be the previous epoch meta block. It has initial 0 values so it can be considered at genesis
+			return &mock.StorerStub{GetCalled: func(key []byte) ([]byte, error) {
+				hdrBytes, _ := json.Marshal(hdrPrevEpochStart)
+				return hdrBytes, nil
+			}}
+		},
+	}
+
+	ec, _ := NewEndOfEpochEconomicsDataCreator(args)
+
+	epochDuration := numberOfSecondsInDay
+	roundsPerEpoch := uint64(epochDuration / roundDur)
+	type testInput struct {
+		blockPerEpochOneShard uint64
+		acFeesInEpoch         *big.Int
```

Comment:
What is ac ? actual? rename it to be better understood

Reply:
renamed.